src/services/IConfiguration.hh

```diff
@@ -0,0 +1,48 @@
+#ifndef RA_SERVICES_ICONFIGURATION
+#define RA_SERVICES_ICONFIGURATION
+#pragma once
+
+#include "ui/WindowViewModelBase.hh"
+
+#include <string>
+
+namespace ra {
+namespace services {
+
+enum class Feature
+{
+    None = 0,
+    Hardcore,
+    Leaderboards,
+    LeaderboardNotifications,
+    LeaderboardCounters,
+    LeaderboardScoreboards,
+};
+
+class IConfiguration {
```

Comment:
The "I" prefix indicates this class is an implementation independent interface. All methods should be pure virtual: "=0".